### ISSUE TITLE
Joystick doesn't appear under Settings > Input Devices > Joystick 1 on macos for ARM (M2) 

### ISSUE THREAD

**Issue by verm71** [86Box/86Box#3451]
**Describe the bug**
The joystick (Logitech Rumblepad 2 USB) appears as connected under the mac's Settings > Game Controller, but when selecting a 2 axis 4 button joystick in 86Box, there are no devices listed under 86Box Input Devices > Joystick 1 > "Devices" drop-down.

**To Reproduce**
1. Connect USB C hub to Macbook Air M2 (this may be irrelevant additional detail)
2. Connect USB Logitech Rumblepad 2 to USB A port on hub
3. Start 86Box, immediately click Settings icon.
4. Go to Input Devices > Joystick > 2 axis 4 button joystick
5. Click Joystick 1
6. Click the Devices drop down.
7. Note that only "None" appears in the list, and no Logitech Rumblepad.

**Expected behavior**
The USB Logitech Rumblepad 2 should appear in the Devices drop-down under Joystick 1.

As a comparison, I installed 86Box in a Windows 11 for ARM VM under Parallels on same Mac, and it successfully lists the Logitech Rumblepad 2 under Devices for the same VM configuration.

**Screenshots**
![image](https://github.com/86Box/86Box/assets/28518082/7e14d86e-b1b9-47f9-9f46-f3f90b401b97)

**Desktop (please complete the following information):**
 - Mac OS Ventura 13.4.1 on a MacBook Air M2
 - 86Box version: v4.0 [build 4689] [352462057] [x86_64, old dynarec]

**Additional context**
Regular build was tried, as was Windows x64, which ran successfully under Rosetta 2 and detected the gamepad.
Gamepad connected through a "5 Create" USB C hub, as the Macbook has no USB A connectors.
Also tried an iBuffalo SNES-like USB gamepad with no success.
I tried with a Bluetooth PS4 controller still connected via USB on the hub (not paired over bluetooth) and it *successfully appears*! This is a good fallback, but would be good to have the other controllers show up too.

**Comment by brunocastello** [comment]
I have the same problem.

MacOS Sonoma 14.4.1 on a Mac Studio M1 Max.
When attempting to bisect, I tested on the following versions: 4.1.1, 4.1, 4.0, 3.1.1, 3.7.1, and 3.4.1, with 3.4.1 being a point release hotfix for the first ever mac version of 86box.

The tests were done with both bluetooth and wired modes of my controller.

My joystick is a 8bitDo SN30 Pro, and I recall having successfully used it once, not sure if I was on Ventura or Monterey, but I do recall that Apple might have done updates to game controllers through this time between Monterey and Sonoma. For example, previously this controller was identified as a PS4 DualShock, now it is correctly identified as an 8bitDo SN30 Pro controller since Ventura.

https://www.dignited.com/108637/8bitdo-controllers-now-officially-supported-by-apple-heres-how-to-setup-on-ios-ipados-macos-and-apple-tv/

Apparently, Apple had issued an update on Ventura 13.2 for game controllers.

My understanding from these tests above is that the code changes for game controllers on macOS might possibly have rendered any code on 86box to use them non-functional.

I will check for a firmware update on my controller, but I very convinced that some Apple update to game controllers on Ventura and later broke it on 86box.

**Comment by josephillips85** [comment]
This problem is unable to replicate on Sonoma with a PS4 controller

<img width="784" alt="image" src="https://github.com/86Box/86Box/assets/28776122/5d22a6de-a27f-40ab-a4c9-356871ef4d72">

**Comment by brunocastello** [comment]
> This problem is unable to replicate on Sonoma with a PS4 controller
> 
> <img alt="image" width="784" src="https://private-user-images.githubusercontent.com/28776122/336704592-5d22a6de-a27f-40ab-a4c9-356871ef4d72.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1OTkzMTEsIm5iZiI6MTcxNzU5OTAxMSwicGF0aCI6Ii8yODc3NjEyMi8zMzY3MDQ1OTItNWQyMmE2ZGUtYTI3Zi00MGFiLWE0YzktMzU2ODcxZWY0ZDcyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA1VDE0NTAxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTViOTY3ODk1YzM5OWQ3N2FjZGU0N2E4ZGU1ZjI1Y2YwZGY2NDM3YjU3ODRmNDMxNDM3NTUwYzA3MjM4ZmJiMzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pTQTs19PdiBgOjR0W5-wiWwBRxPWwGOBCSmoKXYKd40">

It was probably updated since then. I will have another go at it later

**Comment by OBattler** [comment]
I think this can be closed as NOTABUG - clearly Apple had messed up something and it appears to be resolved now.

**Comment by brunocastello** [comment]
I decided to give it another go at it and it's a no go. I wonder if it whatever Apple messed up back then, required changes in the code? The controller (connected through USB) does not show up to be selected. macOS identifies it correctly in System Settings > Game Controllers, but 86Box fails to see it for some reason. 

Considering that previously, other Mac users can see fine other controllers like the original PS4 controller, could it be an issue with identifying this specific controller correctly?

I tested the controller with other games (native and non-native macOS games like ported games) and it's fine. Even Shadow of the Tomb Raider through Steam works with it and identifies it correctly as "8BitDo SN30 Pro Controller". Duke Nukem 3D port sees it and works. I'm going to test it under Windows 11 ARM64 on VMware now.

<img width="1365" height="825" alt="Image" src="https://github.com/user-attachments/assets/7cceca4e-13d3-4fd2-be52-9ab606626129" />

86Box 5.0 (the release version), MacBook Pro M4 Pro, macOS Sequoia 15.6.1.
8bitDo SN30 Pro controller with firmware updated to 2.06 (latest)
